tests/postgres_tests/test_array.py

```diff
@@ -786,3 +786,22 @@ def test_value_omitted_from_data(self):
         self.assertIs(widget.value_omitted_from_data({'field_0': 'value'}, {}, 'field'), False)
         self.assertIs(widget.value_omitted_from_data({'field_1': 'value'}, {}, 'field'), False)
         self.assertIs(widget.value_omitted_from_data({'field_0': 'value', 'field_1': 'value'}, {}, 'field'), False)
+
+
+class TestChoiceFormField(PostgreSQLTestCase):
+    def test_model_field_formfield(self):
+        model_field = ArrayField(models.CharField(max_length=27), choices=(('a1', 'A1'), ('b1', 'B1')))
```

Comment:
I tried saving this model in from admin page, I get the following validation error:

``Select a valid choice. ['a1', 'b1'] is not one of the available choices.``